R/missingLevelsTrain.R

```diff
@@ -0,0 +1,70 @@
+#' @title missingLevelsTrain
+#' @description
+#' Accounts for missing factor levels only present in test data
+#' but not in train data by setting levels to NA
+#' @param fit \cr
+#'   fitted model on training data
+#' @param test.data [`data.frame`] \cr
+#'   data to make predictions for
+#' @return `data.frame` with matching factor levels to fitted model
+#' @keywords internal
+#' @export
+missingLevelsTrain = function(fit, test.data) {
+
+  # https://stackoverflow.com/a/39495480/4185785
+
+  # drop empty factor levels in test data
+  test.data = as.data.frame(droplevels(test.data))
+
+  # 'fit' object structure of 'lm' and 'glmmPQL' is different so we need to
+  # account for it
+  if (any(class(fit) == "glmmPQL")) {
```

Comment:
There are many `if(class(x))` statements. Maybe you can increase use of S3?

Reply:
I'll try 👍

Reply:
see c4d3030. Is this correct?

Reply:
and this bac6533? I replaced the ´class(x)` call and instead created a .default S3 function which basically does nothing for all models except lm*